engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2174,6 +2178,33 @@ public void updateRecoveryCheckpoints(PTOperator operator, UpdateCheckpointsCont
 
   }
 
+  private Collection<PTOperator> getUnifiersInCheckpointGroup(Set<OperatorMeta> checkpointGroup, Collection<PTOperator> operators)
+  {
+    boolean isDelayPartOfGroup = false;
+    for (OperatorMeta om : checkpointGroup) {
+      if (om.getOperator() instanceof Operator.DelayOperator) {
+        isDelayPartOfGroup = true;
+      }
+    }
+    // Include unifiers in group if delay operator is present
+    // Do only if group contains a delay operator
+    if (isDelayPartOfGroup) {
```

Comment:
I think that when logical operators are grouped for checkpointing, then the associated unifiers should also be covered. I would consider that as an omission in the current code. Also, if we agree that this is how it should behave, you can remove the delay checking logic and simply always include unifiers.